### PR TITLE
Update documentation regarding join limitation

### PR DESCRIPTION
**Pull request by affonsov**
### Summary

Update documentation regarding join limitation

### Related Issue

https://bitquill.atlassian.net/browse/AD-401

### Additional Reviewers
@affonsoBQ
@andiem-bq
@alexey-temnikov
@birschick-bq
@mitchell-elholm

### CODE REVIEW

File: src/markdown/sql/sql-limitations.md [hunk 1]
```diff
@@ -53,9 +53,9 @@ Currently, cross collection joins are not supported.
 
 ### Same Collection Joins
 
-Currently, the driver only supports `JOINs` across tables from the same collection as long as we
-are only joining on foreign keys. This is equivalent to presenting the data in its denormalized
-form. For such `JOINs`, the complete set of foreign keys for a table must be used.
+Currently, the driver only supports `JOINs` across tables from the same collection as long as
+the foreign key and primary key relation are present. This is equivalent to presenting the data in its denormalized
+form. For such `JOINs`, the minimal set of foreign keys and their respective primary must be used.
```
Comment (andiemontoyeah) [inline src/markdown/sql/sql-limitations.md:58]: Can we add something like: This means the join condition must have an equality match for each primary key column shared between the tables and no other match conditions.

File: src/markdown/sql/sql-limitations.md [hunk 2]
```diff
@@ -115,14 +115,26 @@ would end up with 4 tables.
...
+#### Table: Minimal Foreign Keys with Primary Keys required
+| Table | Joining Table | FK with PK required |
+| --- | --- | --- |
+| customer | customer_address| customer__id |
+| customer | customer_subscriptions | customer__id |
+| customer | customer_subscriptions_variants | customer__id |
+| customer_address | customer_subscriptions | customer__id |
+| customer_address | customer_subscriptions_variants | customer__id |
+| customer_subscriptions | customer_subscriptions_variants | customer__id and subscriptions_index_lvl_0 | 
+
```
Comment (andiemontoyeah) [inline src/markdown/sql/sql-limitations.md:140]: can we add an example query for customer - customer_subscriptions_variants and customer_address - customer_subscriptions_variants to the list below too?

Comment (andiemontoyeah) [reply src/markdown/sql/sql-limitations.md:140]: and can we add some kind of heading before this list? like "Example queries" or something

File: src/markdown/sql/sql-limitations.md [hunk 3]
```diff
@@ -115,14 +115,26 @@ would end up with 4 tables.
 
 For the tables `customer_address` and `customer_subscriptions` we only need `customer__id`. For `customer_subscriptions_variants`, we need
 `customer__id` and `subscriptions_index_lvl_0`. Between these tables, the following join conditions would be allowed while any others would be rejected.
```
Comment (andiemontoyeah) [inline src/markdown/sql/sql-limitations.md]: - change: "and `subscriptions_index_lvl_0`" to "and possibly `subscriptions_index_lvl_0` depending on the other table being joined".  
- Remove " Between these tables, the following join conditions would be allowed while any others would be rejected."

**Review by andiemontoyeah — COMMENTED** [review]
- on line 111 i made a typo: can you change customer2__id to customer__id? 😅

File: src/markdown/sql/sql-limitations.md [hunk 4]
```diff
@@ -114,15 +118,34 @@ would end up with 4 tables.
...
+| customer_address | customer_subscriptions_variants | customer__id |
+| customer_subscriptions | customer_subscriptions_variants | customer__id and subscriptions_index_lvl_0 | 
+
+#### Table: Minimal Foreign Keys with Primary Keys required in 3 Join Table
+| Table | First Joining Table | FK with PK required for First Join | Second Joining Table | FK with PK required for Second Join |
+| --- | --- | --- | --- | --- |
+| customer | customer_address| customer__id | customer_subscriptions |  customer__id |
+| customer | customer_subscriptions | customer__id | customer_subscriptions_variants | customer__id and subscriptions_index_lvl_0 |
+
+#### Example Queries
```
Comment (alexey-temnikov) [inline src/markdown/sql/sql-limitations.md]: Shall we add example if the JOIN which will not be supported?

File: src/markdown/sql/sql-limitations.md (hunk 1 above)
Comment (affonsov) [reply src/markdown/sql/sql-limitations.md:58]: I added this and also added a written example.

File: src/markdown/sql/sql-limitations.md (hunk 4 above)
Comment (affonsov) [reply src/markdown/sql/sql-limitations.md]: Example added

**Review by andiemontoyeah — APPROVED** [review]

**Review by birschick-bq — APPROVED** [review]

**Pull request merged**